codebuild/bin/install_ubuntu_dependencies.sh

```diff
@@ -39,9 +39,10 @@ base_packages() {
   apt install -y software-properties-common
   add-apt-repository ppa:ubuntu-toolchain-r/test -y
   add-apt-repository ppa:longsleep/golang-backports -y
+  add-apt-repository ppa:deadsnakes/ppa -y # needed for python3.9 and python3.9-distutils
   apt-get update -o Acquire::CompressionTypes::Order::=gz
 
-  DEPENDENCIES="unzip make indent iproute2 kwstyle libssl-dev net-tools  tcpdump valgrind lcov m4 nettle-dev nettle-bin pkg-config psmisc gcc g++ zlibc zlib1g-dev python3-pip python3-testresources llvm curl shellcheck git tox cmake libtool ninja-build golang-go quilt jq"
+  DEPENDENCIES="unzip make indent iproute2 kwstyle libssl-dev net-tools  tcpdump valgrind lcov m4 nettle-dev nettle-bin pkg-config psmisc gcc g++ zlibc zlib1g-dev python3-pip python3-testresources llvm curl shellcheck git tox cmake libtool ninja-build golang-go quilt jq python3.9 python3.9-distutils"
```

Comment:
is the usual `apt-get install python3.9` sufficient or do you also need to add it to the `$PATH`?

Reply:
I didn't need to add it to $PATH

Reply:
The [CodeBuild image](https://github.com/aws/aws-codebuild-docker-images/blob/master/ubuntu/standard/4.0/Dockerfile#L267) already has python3.9, instead of installing it twice, can this be moved to `install_test_dependencies.sh` (which isn't used in CI) or check for it's existence first ?

Reply:
Would apt-get reinstall it if it's already installed?

Reply:
`apt` only knows about deb packages, so if python was setup with `pyenv`, apt will install it a second time.  I vetted that you can have both install methods on the same host, but this speaks to a larger issue where our CI doesn't match our devdesktop setups.

Reply:
updated to check for python3.9 already being installed